EasyPost/Utilities/Cryptography.cs

```diff
@@ -0,0 +1,152 @@
+
+            return bytes.AsHexString();
+        }
+
+        /// <summary>
+        ///     Convert a byte array to a string using a specific encoding (defaults to UTF-8)
+        /// </summary>
+        /// <param name="bytes">Byte array to convert to string.</param>
+        /// <param name="encoding">Encoding to use. Default: UTF-8</param>
+        /// <returns>String</returns>
+        public static string AsString(this byte[] bytes, Encoding? encoding = null)
+        {
+            encoding ??= Encoding.UTF8;
+
+            return encoding.GetString(bytes);
+        }
+
+        /// <summary>
+        ///     Calculate the HMAC-SHA256 hex digest of a byte array.
+        /// </summary>
+        /// <param name="data">Data to calculate hex digest for.</param>
+        /// <param name="secret">Key used to calculate data hex digest.</param>
+        /// <param name="normalizationForm">Normalization type to use when normalizing key. Default: No normalization.</param>
+        /// <returns>Hex digest of data.</returns>
+        public static string CalculateHMACSHA256HexDigest(this byte[] data, string secret, NormalizationForm? normalizationForm = null)
+        {
+            if (normalizationForm != null)
+            {
+                secret = secret.Normalize(normalizationForm.Value);
+            }
+
+            byte[] keyBytes = Encoding.UTF8.GetBytes(secret);
+
+            using HMACSHA256 hmac = new HMACSHA256(keyBytes);
+            byte[] hash = hmac.ComputeHash(data);
+
+            return hash.AsHexString();
+        }
+
+        /// <summary>
+        ///     Check whether two signatures match. This is safe against timing attacks.
+        /// </summary>
+        /// <param name="signature1">First signature.</param>
+        /// <param name="signature2">Second signature.</param>
+        /// <returns>Whether the two signatures match.</returns>
+        public static bool SignaturesMatch(byte[] signature1, byte[] signature2)
+        {
+            // short-circuit if signatures are not the same length
+            if (signature1.Length != signature2.Length)
+            {
+                return false;
+            }
+
+            bool err = false;
+            for (int i = 0; i < signature1.Length; i++)
+            {
+                if (signature1[i] != signature2[i])
+                {
+                    err = true;
+                }
+            }
+
+            return !err;
+        }
+
+        /// <summary>
+        ///     Check whether two signatures match. This is safe against timing attacks.
+        /// </summary>
+        /// <param name="signature1">First signature.</param>
+        /// <param name="signature2">Second signature.</param>
+        /// <returns>Whether the two signatures match.</returns>
+        public static bool SignaturesMatch(string signature1, string signature2)
+        {
+            signature1 = signature1.ToLower();
+            signature2 = signature2.ToLower();
```

Comment:
Issue: I don't foresee a need for us to be lowercasing here. That lowers the integrity of the check. Let's remove this completely and see if the tests pass if they do, if so we don't need this.